<a href="https://colab.research.google.com/github/kimchaeyoung-student/Kaggle/blob/master/Titanic_tutorial.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Data Camp tutorial**
### **<타이타닉호 승객 데이터 분석>**

In [0]:
#pandas 라이브러리 불러오기

import pandas as pd #as pd를 통해 pandas 를 pd로 줄여쓸 수 있음

In [0]:
train_url = "http://s3.amazonaws.com/assets.datacamp.com/course/Kaggle/train.csv"
train = pd.read_csv(train_url)

test_url = "http://s3.amazonaws.com/assets.datacamp.com/course/Kaggle/test.csv"
test = pd.read_csv(test_url)

#pandas에서 제공하는 DataFrame 형태로 데이터가 저장됨

In [5]:
train.describe()  #수치 데이터만 요약해서 보여줌

PassengerId    Survived      Pclass  ...       SibSp       Parch        Fare
count   891.000000  891.000000  891.000000  ...  891.000000  891.000000  891.000000
mean    446.000000    0.383838    2.308642  ...    0.523008    0.381594   32.204208
std     257.353842    0.486592    0.836071  ...    1.102743    0.806057   49.693429
min       1.000000    0.000000    1.000000  ...    0.000000    0.000000    0.000000
25%     223.500000    0.000000    2.000000  ...    0.000000    0.000000    7.910400
50%     446.000000    0.000000    3.000000  ...    0.000000    0.000000   14.454200
75%     668.500000    1.000000    3.000000  ...    1.000000    0.000000   31.000000
max     891.000000    1.000000    3.000000  ...    8.000000    6.000000  512.329200

[8 rows x 7 columns]

In [6]:
train.shape #해당 데이터셋의 행과 열이 몇개인지 알려줌

(891, 12)

### **사람의 생존율**

In [7]:
train["Survived"].value_counts()

#value_counts() 함수를 호출하면 절대값으로 결과를 돌려줌

#Survived == 0 : 사람이 죽었다.
#Survived == 1 : 사람이 살았다.

0    549
1    342
Name: Survived, dtype: int64

In [8]:
train["Survived"].value_counts(normalize = True)

#normalize 옵션을 True로 설정하면 퍼센트(%)로 결과를 리턴

0    0.616162
1    0.383838
Name: Survived, dtype: float64

### **성별에 따른 생존율**

In [9]:
train['Survived'][train['Sex'] == 'male'].value_counts(normalize=True)

#train["Sex"] == 'male' : train 데이터 개수만큼 boolean vector가 리턴됨.
#값은 해당 데이터의 성별이 남성인 데이터는 True, 아니면 False를 가짐.
#이 boolean vector로 indexing을 하게 되면 True 데이터들만 indexing 할 수 있음.

0    0.811092
1    0.188908
Name: Survived, dtype: float64

In [10]:
train['Survived'][train['Sex'] == 'female'].value_counts(normalize=True)

1    0.742038
0    0.257962
Name: Survived, dtype: float64

남성 생존율 : 약 19%, 여성 생존율 : 약 74%

**남자는 무조건 죽고, 여자는 무조건 산다**고 예측한 것이 **모두 죽는다**고 예측한 정확도 보다 높다.

(모두 죽는다고 예측했을 때의 정확도 : 약 62%)

### **나이에 따른 생존율**

(18세 미만과 이상으로 나누어서 분석)

In [0]:
train["Child"] = float('NaN')

#train data set에 ‘Child’라는 열이 생성되고, 값은 전부 NaN으로 채워짐

In [28]:
train["Child"][train['Age'] < 18] = 1
train["Child"][train['Age'] >= 18] = 0

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


(위에서 NaN이 그대로 남아있는 경우는 해당 승객의 Age값이 존재하지 않았기 때문.

데이터 분석을 하다보면 위와 같이 없는 데이터가 많으므로, 단순히 boolean으로 표현하면 안됨!)

In [29]:
print(train["Survived"][train["Child"] == 1].value_counts(normalize = True))

#18세 미만의 생존율

1    0.539823
0    0.460177
Name: Survived, dtype: float64


In [30]:
print(train["Survived"][train["Child"] == 0].value_counts(normalize = True))

#18세 이상의 생존율

0    0.618968
1    0.381032
Name: Survived, dtype: float64


18세 미만의 생존율 : 54%

18세 이상의 생존율 : 38%

성별에 따른 생존율보다는 적은 차이를 보인다.